File: common/common-utils/src/main/java/com/leopoldhsing/digitalhippo/common/mapper/product/ProductImageMapper.java
package com.leopoldhsing.digitalhippo.common.mapper.product;

import com.leopoldhsing.digitalhippo.model.entity.Product;
import com.leopoldhsing.digitalhippo.model.entity.ProductImage;
import com.leopoldhsing.digitalhippo.model.enumeration.ProductFileType;
import com.leopoldhsing.digitalhippo.model.vo.ProductImageVo;
import org.springframework.beans.BeanUtils;

public class ProductImageMapper {
    private ProductImageMapper() {
    }

    public static ProductImage mapToProductImage(ProductImageVo productImageVo, Product product) {
        ProductImage productImage = new ProductImage();

        BeanUtils.copyProperties(productImageVo, productImage);
        productImage.setFileType(ProductFileType.fromValue(productImageVo.getFileType()));
        productImage.setProduct(product);

        return productImage;
    }

    public static ProductImageVo mapToProductImageVo(ProductImage productImage) {
        ProductImageVo productImageVo = new ProductImageVo();

        BeanUtils.copyProperties(productImage, productImageVo);
        productImageVo.setFileType(productImage.getFileType().getValue());

        return productImageVo;
    }
}

File: service/product/src/main/kotlin/com/leopoldhsing/digitalhippo/product/mapper/ProductImageMapper.kt
package com.leopoldhsing.digitalhippo.product.mapper

import com.leopoldhsing.digitalhippo.model.entity.Product
import com.leopoldhsing.digitalhippo.model.entity.ProductImage
import com.leopoldhsing.digitalhippo.model.enumeration.ProductFileType
import com.leopoldhsing.digitalhippo.model.vo.ProductImageVo
import org.springframework.stereotype.Component

@Component
class ProductImageMapper {
    companion object {
        fun mapTopProductImage(productImageVo: ProductImageVo, product: Product): ProductImage {
            val productImage = ProductImage()
            productImage.url = productImageVo.url
            productImage.payloadId = productImageVo.payloadId
            productImage.filename = productImageVo.filename
            productImage.filesize = productImageVo.filesize
            productImage.height = productImageVo.height
            productImage.width = productImageVo.width
            productImage.mimeType = productImageVo.mimeType
            productImage.fileType = ProductFileType.fromValue(productImageVo.fileType)
            productImage.product = product

            return productImage
        }
    }
}